Vagrantfile

```diff
@@ -12,6 +12,69 @@ Vagrant.configure("2") do |config|
   # using a specific IP.
   config.vm.network "private_network", ip: "192.168.127.127"
 
+
+  # vb.customize does not returns output from the command
+  id_file = ".vagrant/machines/default/virtualbox/id"
+  # don't know the id, can't search for the controller name
+  # using the default on the config.vm.box and hoping for the best
+  floppy_controller = "Floppy Controller Controller"
+  has_cdrom = false
+
+  # doesn't exist until VM is provisioned the first time
+  if FileTest.exist?(id_file) and FileTest.readable?(id_file)
+    # unsetting, if nothing is found no need to remove again
+    floppy_controller = ""
+    id = File.read(id_file)
+    output = `VBoxManage showvminfo #{id}`
+    previous_line = ''
+
+    for line in output.each_line
+      if line =~ /I82078/
+        floppy_controller = (previous_line.gsub(/\s/, '').split(':'))[1]
+      elsif line =~ /^SATA\sController\s\(1,\s0\)\:\sEmpty/
+          has_cdrom = true
+      else
+        previous_line = line
+      end
+    end
+  end
+
+  config.vm.provider "virtualbox" do |vb|
+     # Display the VirtualBox GUI when booting the machine
+     # Might be useful, but not required, so leaving commented
+     # vb.gui = true
+     # Customize the amount of memory on the VM:
+     vb.memory = "1024"
+     # Giving a nice name to it
+     vb.name = "CPAN Testers - Development"
+     # see https://www.virtualbox.org/manual/ch08.html
+     # configure the minimum memory for video (set to 8MB originally)
+     vb.customize ["modifyvm", :id, "--vram", "9"]
+     # This needs to be set as idempotent change
+     # remove the floppy
+     unless floppy_controller.empty?
+       vb.customize ["storagectl", :id, "--name", floppy_controller, "--remove"]
+     end
+     # adds a CDROM
+     # also assumes config.vm.box has a single SATA HD in port 0
+     unless has_cdrom
+       vb.customize ["storageattach", :id, "--storagectl", "SATA Controller", "--port", "1", "--medium", "emptydrive"]
+     end
+  end
+
+  # Guest Additions handling
+  config.vm.provision "shell", inline: "apt-get update;apt-get -y upgrade;apt-get -y install module-assistant;m-a prepare --text-mode --non-inter"
```

Comment:
Besides, we could use the "auto-upgrade" to even validate if the newest packages from Debian would cause any problem in the production server. I would need to compare packages installed by default on the production server and the VM in the Vagrant box we are using to understand the differences. If there is any update being executed on the production server, there is a change that the box is outdated anyway, so the conditions for testing wouldn't be the same.
There is also de apt-get "pin" feature, but I'm not sure to use it to guarantee specific packages versions. I'll look for this and see if it is possible to reproduce the production environment in terms of packages version directly in the VM in a automatic fashion.